### PR TITLE
feat : Adding new icons

### PR DESCRIPTION
**Pull request by Brijeshkrishna**
feat: Updated folder icon with rounded corners and new icons

### CODE REVIEW

**Comment by Brijeshkrishna** [comment]
feat: Added new icons, changed folder icons with rounded corners.

Screenshot_20230919_192001-1

**Review by cafkafk — CHANGES_REQUESTED** [review]
We've discussed the folder icon in the past https://github.com/eza-community/eza/pull/137. I suggest you split this up into multiple PRs, one for the folder icon change, which is bound to require a bit of discussion, and one for the other icons, which should be much quicker.

**Comment by Brijeshkrishna** [comment]
Reverted

**Review by cafkafk — CHANGES_REQUESTED** [review]
I tried to rebase your commits locally to change the summaries so the test pass, but I don't have permission to do that. You'll have to change them, and they should be squashed into a single commit

**Review by cfxegbert — COMMENTED** [review]
Sorry about being a bit nit picky here.  Some of these comments about which icon we should use I need to address and will get back to you.

File: src/output/icons.rs [hunk 1]
```diff
@@ -396,6 +402,7 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "ex"             => Icons::LANG_ELIXIR,      // 
     "exe"            => Icons::OS_WINDOWS_CMD,   // 
     "exs"            => Icons::LANG_ELIXIR,      // 
+    "fdmdownload"    => '\u{f0965}',             // 󰥥
```
[the remark this answers is not in the document]
Comment (cfxegbert) [reply src/output/icons.rs:405]: Can we change this to use the same icon as "download".  The simpler icon reads better at small sizes.

File: src/output/icons.rs [hunk 2]
```diff
@@ -61,7 +61,7 @@ impl Icons {
     const LANG_R: char          = '\u{f25d}';  // 
     const LANG_RUBY: char       = '\u{e21e}';  // 
     const LANG_RUBYRAILS: char  = '\u{e73b}';  // 
-    const LANG_RUST: char       = '\u{e7a8}';  // 
+    const LANG_RUST: char       = '\u{e68b}';  // 
```
[the remark this answers is not in the document]
Comment (Brijeshkrishna) [reply src/output/icons.rs:64]: e68b is bigger than e7a8 , e68b is normal size compare to other icons

[the remark this answers is not in the document]
Comment (cfxegbert) [reply src/output/icons.rs:64]: Good catch.  Can you update all the icon comments for LANG_RUST in the file?

File: src/output/icons.rs [hunk 3]
```diff
@@ -314,9 +319,10 @@ const EXTENSION_ICONS: Map<&'static str, char> = phf_map! {
     "avif"           => Icons::IMAGE,            // 
     "avro"           => Icons::JSON,             // 
     "awk"            => Icons::SHELL_CMD,        // 
-    "bash"           => Icons::SHELL_CMD,        // 
+    "bash"           => '\u{ebca}',              // 
```
[the remark this answers is not in the document]
Comment (Brijeshkrishna) [reply src/output/icons.rs:326]: since bash icon is available.

[the remark this answers is not in the document]
Comment (cfxegbert) [reply src/output/icons.rs:326]: We would rather stay consistent with the other shell icons.

File: src/output/icons.rs [hunk 4]
```diff
@@ -131,13 +135,12 @@ const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
...
     "npm_cache"           => Icons::FOLDER_NPM,     // 
-    "pam.d"               => Icons::FOLDER_KEY,     // 󰢬
     "Pictures"            => '\u{f024f}',           // 󰉏
     "ssh"                 => Icons::FOLDER_KEY,     // 󰢬
-    "sudoers.d"           => Icons::FOLDER_KEY,     // 󰢬
     "Videos"              => '\u{f03d}',            // 
-    "xbps.d"              => Icons::FOLDER_CONFIG,  // 
-    "xorg.conf.d"         => Icons::FOLDER_CONFIG,  // 
+    "Contacts"            => '\u{f024c}',           // 󰉌
+    "Favorite"            => '\u{f069d}',           // 󰚝
```
[the remark this answers is not in the document]
Comment (devinherron) [reply src/output/icons.rs]: Windows 10 and 11 have "Contacts" and "Favorites" (plural)

[the remark this answers is not in the document]
Comment (Brijeshkrishna) [reply src/output/icons.rs]: Yes, and if the user created this folder on Linux, let it support it.

File: src/output/icons.rs [hunk 5]
```diff
@@ -131,13 +135,12 @@ const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
...
+    "Drive"               => '\u{f024e}',           // 󰉎
```
[the remark this answers is not in the document]
Comment (Brijeshkrishna) [reply src/output/icons.rs]: No, changed to 🖴 U+1F5B4

File: src/output/icons.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (PThorpe92) [reply src/output/icons.rs]: I agree 'Favorites' needs to be plural

File: src/output/icons.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (cfxegbert) [reply src/output/icons.rs]: 1f5b4 is not a codepoint in Nerdfonts

[the remark this answers is not in the document]
Comment (Brijeshkrishna) [reply src/output/icons.rs]: Yes, it is not a nerd font, but it is supported by all fonts (it is in the Unicode Miscellaneous Symbols and Pictographs block).

[the remark this answers is not in the document]
Comment (cfxegbert) [reply src/output/icons.rs]: It is not supported by any of the NerdFonts I have tested.

[the remark this answers is not in the document]
Comment (cfxegbert) [reply src/output/icons.rs]: Code I used to test on my mac

```python
from pathlib import Path
from fontTools.ttLib import TTFont
import unicodedata

def find_fonts():
    for font_dir in ["/System/Library/Fonts", "/Library/Fonts", "~/Library/Fonts"]:
        for path in Path(font_dir).expanduser().rglob("*.ttf"):
            yield str(path)


def char_in_font(unicode_char, font):
    for cmap in font['cmap'].tables:
        if cmap.isUnicode():
            if ord(unicode_char) in cmap.cmap:
                return True
    return False

def test_char(char):
    for fontpath in find_fonts():
        font = TTFont(fontpath)
        if char_in_font(char, font):
            # name = unicodedata.name(char)
            # print(f"{char} {name} in {fontpath}")
            print(f"{char} in {fontpath}")

#test_char("\U000f024e")
test_char("\U0001f5b4")
```

[the remark this answers is not in the document]
Comment (cfxegbert) [reply src/output/icons.rs]: Looking at the table https://en.wikipedia.org/wiki/Miscellaneous_Symbols_and_Pictographs there are large gaps at the end of the table starting at U+1F54x

File: src/output/icons.rs
Comment (MartinFillon) [inline src/output/icons.rs:1]: I am also going to be picky but do we think about giving all icons a constant in the Icons enum ?

File: src/output/icons.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (Brijeshkrishna) [reply src/output/icons.rs]: Ok then, removed.

File: src/output/icons.rs
Comment (cfxegbert) [reply src/output/icons.rs:1]: I'll do that in a different pull request.

**Review by cfxegbert — APPROVED** [review]

**Comment by cafkafk** [comment]
I'll take a look

**Review by cafkafk — CHANGES_REQUESTED** [review]
You'll have to rebase and fix the commit summaries, so they conform to conventional commits. Else, testing this locally looks good. Nice work!

File: src/output/icons.rs [hunk 6]
```diff
@@ -173,9 +179,10 @@ const FILENAME_ICONS: Map<&'static str, char> = phf_map! {
     ".npmrc"              => Icons::NPM,            // 
     ".profile"            => Icons::SHELL,          // 󱆃
     ".python_history"     => Icons::LANG_PYTHON,    // 
-    ".rustfmt.toml"       => Icons::LANG_RUST,      // 
+    ".rustfmt.toml"       => Icons::LANG_RUST,      // 
     ".rvm"                => Icons::LANG_RUBY,      // 
     ".rvmrc"              => Icons::LANG_RUBY,      // 
+    "README"              => Icons::README,         // 󰂺
```
Comment (cfxegbert) [inline src/output/icons.rs:185]: Just noticed this is not sorted properly.  You can fix or I can fix the next time I edit the icons.

**Comment by cfxegbert** [comment]
# Added extensions
| Ext           | Icon |     Class      |  Code   | Notes |
|---------------|:----:|:--------------:|:-------:|-------|
| `bdf`         |     |   nf-fa-font   | \uf031  |       |
| `fdmdownload` |  󰇚  | nf-md-download | \uf01da |       |
| `lib`         |     | nf-cod-library | \ueb9c  |       |
| `psf`         |     |   nf-fa-font   | \uf031  |       |
| `readme`      |  󰂺  |   nf-md-book   | \uf00ba |       |

# Changed extensions
| Ext     | Icon |     Class      |  Code  | Old icon |    Old class     | Old code | Notes                                                    |
|---------|:----:|:--------------:|:------:|:--------:|:----------------:|:--------:|----------------------------------------------------------|
| `dll`   |     | nf-cod-library | \ueb9c |         |  nf-fa-windows   |  \uf17a  |                                                          |
| `r`     |     |   nf-seti-r    | \ue68a |         | nf-fa-registered |  \uf25d  | Change from generic registered symbol to R language icon |
| `rdata` |     |   nf-seti-r    | \ue68a |         | nf-fa-registered |  \uf25d  | Change from generic registered symbol to R language icon |
| `rds`   |     |   nf-seti-r    | \ue68a |         | nf-fa-registered |  \uf25d  | Change from generic registered symbol to R language icon |
| `rlib`  |     |  nf-seti-rust  | \ue68b |         |   nf-dev-rust    |  \ue7a8  | Same icon with bigger size                               |
| `rmeta` |     |  nf-seti-rust  | \ue68b |         |   nf-dev-rust    |  \ue7a8  | Same icon with bigger size                               |
| `rs`    |     |  nf-seti-rust  | \ue68b |         |   nf-dev-rust    |  \ue7a8  | Same icon with bigger size                               |

# Removed extensions
| Ext | Notes |
|-----|-------|

# Added files
| Name        | Icon |        Class         |  Code   | Notes                          |
|-------------|:----:|:--------------------:|:-------:|--------------------------------|
| `Contacts`  |  󰉌  | nf-md-folder_account | \uf024c | Contacts directory on Windows  |
| `Favorites` |  󰚝  |  nf-md-folder_star   | \uf069d | Favorites directory on Windows |
| `home`      |  󱂵  |  nf-md-folder_home   | \uf10b5 | Home directory                 |
| `README`    |  󰂺  |      nf-md-book      | \uf00ba |                                |

# Changed files
| Name            | Icon |    Class     |  Code  | Old icon |  Old class  | Old code | Notes                      |
|-----------------|:----:|:------------:|:------:|:--------:|:-----------:|:--------:|----------------------------|
| `.rustfmt.toml` |     | nf-seti-rust | \ue68b |         | nf-dev-rust |  \ue7a8  | Same icon with bigger size |
| `Cargo.lock`    |     | nf-seti-rust | \ue68b |         | nf-dev-rust |  \ue7a8  | Same icon with bigger size |
| `Cargo.toml`    |     | nf-seti-rust | \ue68b |         | nf-dev-rust |  \ue7a8  | Same icon with bigger size |
| `release.toml`  |     | nf-seti-rust | \ue68b |         | nf-dev-rust |  \ue7a8  | Same icon with bigger size |

# Removed files
| Name | Notes |
|------|-------|

**Review by cafkafk — CHANGES_REQUESTED** [review]
Seems like you have a formatting issue:

```
treefmt-check> [INFO]: #rustfmt: 61 files processed in 1.27s
treefmt-check> traversed 201 files
treefmt-check> matched 104 files to formatters
treefmt-check> left with 104 files after cache
treefmt-check> of whom 1 files were re-formatted
treefmt-check> all of this in 1s
treefmt-check> On branch main
treefmt-check> Changes not staged for commit:
treefmt-check>   (use "git add <file>..." to update what will be committed)
treefmt-check>   (use "git restore <file>..." to discard changes in working directory)
treefmt-check>  modified:   src/output/icons.rs
treefmt-check> no changes added to commit (use "git add" and/or "git commit -a")
treefmt-check> diff --git a/src/output/icons.rs b/src/output/icons.rs
treefmt-check> index 7242ea4..c13a3c8 100644
treefmt-check> --- a/src/output/icons.rs
treefmt-check> +++ b/src/output/icons.rs
treefmt-check> @@ -131,7 +131,7 @@ const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
treefmt-check>      "etc"                 => Icons::FOLDER_CONFIG,  // 
treefmt-check>      "Favorites"           => '\u{f069d}',           // 󰚝
treefmt-check>      "hidden"              => Icons::FOLDER_HIDDEN,  // 󱞞
treefmt-check> -    "home"                => '\u{f10b5}',           // 󱂵 
treefmt-check> +    "home"                => '\u{f10b5}',           // 󱂵
treefmt-check>      "include"             => Icons::FOLDER_CONFIG,  // 
treefmt-check>      "Mail"                => '\u{f01f0}',           // 󰇰
treefmt-check>      "Movies"              => '\u{f0fce}',           // 󰿎
treefmt-check> @@ -145,7 +145,7 @@ const DIRECTORY_ICONS: Map<&'static str, char> = phf_map! {
treefmt-check>      "Videos"              => '\u{f03d}',            // 
treefmt-check>      "xbps.d"              => Icons::FOLDER_CONFIG,  // 
treefmt-check>      "xorg.conf.d"         => Icons::FOLDER_CONFIG,  // 
treefmt-check> -    
treefmt-check> +
treefmt-check>  };
treefmt-check>  
treefmt-check>  /// Mapping from full filenames to file icon. This mapping should also contain
```

**Review by cafkafk — APPROVED** [review]
And that's it. Integration tests pass, CI passes, everything seems to work. Thanks for your contribution @Brijeshkrishna, LGTM :heart:

**Pull request merged**